l10n_it_delivery_note/security/ir_rule.xml

```diff
@@ -13,4 +13,25 @@
         </field>
     </record>
 
+    <record id="stock_delivery_note_personal_rule" model="ir.rule">
+        <field name="name">Personal Delivery note</field>
+        <field name="model_id" ref="l10n_it_delivery_note.model_stock_delivery_note" />
+        <field name="domain_force">
+        ['|','|',('partner_id','=',user.partner_id.id),('partner_id','=',user.partner_id.parent_id.id),'|',('partner_shipping_id','=',user.partner_id.id),('partner_shipping_id','=',user.partner_id.parent_id.id)]
+        </field>
+        <field name="groups" eval="[(4, ref('base.group_portal'))]" />
+    </record>
+
+    <record id="stock_delivery_note_line_personal_rule" model="ir.rule">
+        <field name="name">Personal Delivery note line</field>
+        <field
+            name="model_id"
+            ref="l10n_it_delivery_note.model_stock_delivery_note_line"
+        />
+        <field name="domain_force">
+        ['|','|',('delivery_note_id.partner_id','=',user.partner_id.id),('delivery_note_id.partner_id','=',user.partner_id.parent_id.id),'|',('delivery_note_id.partner_shipping_id','=',user.partner_id.id),('delivery_note_id.partner_shipping_id','=',user.partner_id.parent_id.id)]
+        </field>
+        <field name="groups" eval="[(4, ref('base.group_portal'))]" />
+    </record>
```

Comment:
Hai valutato l'uso di `commercial_partner_id`? Vedi ad esempio le regole sui SO

https://github.com/OCA/OCB/blob/14.0/addons/sale/security/sale_security.xml#L91

Mi sembra una soluzione più robusta, ma magari non è adatta a questo contesto.

Reply:
Fatto

Reply:
purtroppo ha fallito però i test
in teoria doveva funzionare con commercial_partner, adesso faccio il debug

Reply:
> 

@aleuffre ho fatto vari test e le regole che ho impostato inizialmente sono più inclusive rispetto all'uso di `commercial_partner` perchè utilizzando `commercial_partner` si considera sempre l'azienda, mentre io avevo incluso anche le persone siccome nelle DN è possibile impostare anche quelle nel partner

Reply:
@aleuffre rettifico il commento precedente, avevi ragione consigliandomi di usare `commercial_partner_id` per uniformare le regole ai SO, ma dovevo sistemarle un po' per adattarle siccome non mi sembra che nelle DN vengano aggiunti i followers nello stesso modo che nei SO, perciò ho usato `partner_id` per il confronto invece che `message_partner_ids`.